src/test/client/TestClient.h

```diff
@@ -86,6 +93,128 @@ class ClientScaffold : public Client {
       wait_on_list(waiting_for_reclaim);
       return session->reclaim_state == MetaSession::RECLAIM_FAIL ? true : false;
     }
+
+    void write_n_bytes_async(struct Fh *fh, size_t to_write, size_t block_size,
+                            int iov_cnt, off_t *offset) {
+      /// @brief Write N bytes of data asynchronously.
+      /// @param fh - File handle
+      /// @param to_write - bytes to write
+      /// @param block_size - Total size of each iovec structs array
+      /// @param iov_cnt - Number of elements in iovec structs array
+      /// @param offset - location to start writing from
+
+      const size_t DATA_PER_BLOCK = size_t(block_size / iov_cnt);
+      // to persist memory address of out_buf after every loop iteration
+      std::vector<std::unique_ptr<char[]>> buffers;
+      buffers.reserve(iov_cnt);
+      struct iovec iov_buf_out[iov_cnt];
+
+      for (int i = 0; i < iov_cnt; ++i) {
+        auto out_buf = std::make_unique<char[]>(DATA_PER_BLOCK);
+        memset(out_buf.get(), 'a' + i, DATA_PER_BLOCK);
+        iov_buf_out[i].iov_base = out_buf.get();
+        iov_buf_out[i].iov_len = DATA_PER_BLOCK;
+        buffers.push_back(std::move(out_buf));
+      }
+
+      std::unique_ptr<C_SaferCond> writefinish = nullptr;
```

Comment:
Is it really necessary to use unique_ptr here, since the executing context is blocked using a safer cond?

Reply:
this has been the case everywhere in `nonblocking.cc`, it doesn't do any harm as such, does it?

Reply:
done https://github.com/ceph/ceph/compare/1ce7a85da0c16b4d2a28bb0cc3c229e223fd7090..eac7652a52532eee193f14355a05ddee5ef767ad